runtime/src/snapshot_utils.rs

```diff
@@ -65,8 +66,19 @@ use {
 
 mod archive_format;
 mod snapshot_storage_rebuilder;
+use crate::accounts_db::AppendVecId;
 pub use archive_format::*;
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum SnapshotFrom {
+    /// Build from the snapshot archive
+    Archive,
+    /// Build directly from the files in the snapshots/ and anccounts/
+    File,
+    /// No snapshot.  Genesis should be used
+    None,
```

Comment:
Isn't it clearer to just have an `Option<SnapshotFrom>` where `Some(x)` means we're loading either from files or archives, and `None` means we don't load from snapshot, i.e. genesis?

Reply:
I want to handle 3 possibilities here, valid1, valid2, error.  match handles them with out of them being None.

If I use Option, do I have to split the logic into 2 layers, as below?
If Some(x) {
  match x {
    valid1 => {...},
    valid2 => {...}
}}

Reply:
Never mind.  I see how it can be done now.  I should use Some(valid1), Some(valid2) and None as the match options.